### PR TITLE
feat: able to offline worker instead of remove

### LINKED ISSUE #393: should allow worker to be offlined instead of removed  to keep its model uid
image

### PR DESCRIPTION
**Pull request by bjwswang**
#### What type of PR is this?
/kind feat

#### Which issue(s) this PR fixes

Fixes #393 

#### Special notes for your reviewer
1. Add a new condition reason `Offline` to worker
2. Add a new field `replicas`(0,1) which is same as `deployment.spec.replicas`
3. Watch `Pod` instead of `Deployment` due to replica changes won't change deployment status
4. Create LLM/Embedder at `BeforeStart` which means `llm/embedder`  will be created even worker is not ready.

### CODE REVIEW

File: api/base/v1alpha1/worker.go [hunk 1]
```diff
@@ -34,6 +34,10 @@ const (
 
 const (
 	LabelWorkerType = Group + "/worker-type"
+
+	// Labels for worker's Pod
+	WorkerPodSelectorLabel = "app.kubernetes.io/name"
+	WorkerPodLabel         = Group + "/worker"
```
Comment (bjwswang) [inline api/base/v1alpha1/worker.go:40]: Use this label to filter pods that belongs to a worker

Comment (bjwswang) [reply api/base/v1alpha1/worker.go:40]: Set owner reference in `PodTemplateSpec` will be overrided

File: api/base/v1alpha1/worker.go [hunk 2]
```diff
@@ -105,6 +109,27 @@ func (worker Worker) ReadyCondition() Condition {
 	}
 }
 
+func (worker Worker) OfflineCondition() Condition {
```
Comment (bjwswang) [inline api/base/v1alpha1/worker.go:112]: This is a new condition for `Offline`

File: controllers/worker_controller.go [hunk 3]
```diff
@@ -287,17 +221,19 @@ func (r *WorkerReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
-				UpdateFunc: func(ue event.UpdateEvent) bool {
-					oldDep := ue.ObjectOld.(*appsv1.Deployment)
-					newDep := ue.ObjectNew.(*appsv1.Deployment)
-					return !reflect.DeepEqual(oldDep.Status, newDep.Status)
-				},
-			},
-		)).
+		Watches(&source.Kind{Type: &corev1.Pod{}}, handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+			pod := o.(*corev1.Pod)
+			if pod.Labels != nil && pod.Labels[arcadiav1alpha1.WorkerPodLabel] != "" {
```
Comment (bjwswang) [inline controllers/worker_controller.go:226]: filter pods based on Label

**Review by Abirdcfly — APPROVED** [review]

**Pull request merged**